src/main/java/frc/robot/subsystems/pivot/PivotIOSparkMax.java

```diff
@@ -0,0 +1,87 @@
+package frc.robot.subsystems.pivot;
+
+import static frc.robot.Constants.PivotConstants.*;
+
+import com.revrobotics.CANSparkLowLevel;
+import com.revrobotics.CANSparkMax;
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.math.controller.ArmFeedforward;
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.trajectory.ExponentialProfile;
+import edu.wpi.first.wpilibj.DutyCycleEncoder;
+
+public class PivotIOSparkMax implements PivotIO {
+  private boolean openLoop = false;
+  private double appliedVolts = 0.0;
+  private final CANSparkMax leftMotor =
+      new CANSparkMax(LEFT_MOTOR_PORT, CANSparkLowLevel.MotorType.kBrushless);
+  private final CANSparkMax rightMotor =
+      new CANSparkMax(RIGHT_MOTOR_PORT, CANSparkLowLevel.MotorType.kBrushless);
+  private final DutyCycleEncoder absoluteEncoder = new DutyCycleEncoder(ENCODER_PORT);
+  private final PIDController controller = new PIDController(50, 0.0, 0.0);
+  private final ArmFeedforward pivotFFModel = new ArmFeedforward(0.0, 0.0379, 5.85, 0.04);
+  private final ExponentialProfile pivotProfile =
+      new ExponentialProfile(
+          ExponentialProfile.Constraints.fromCharacteristics(10, pivotFFModel.kv, pivotFFModel.ka));
+  private ExponentialProfile.State pivotGoal;
+  private ExponentialProfile.State pivotSetpoint;
+
+  public PivotIOSparkMax() {
+    pivotGoal = new ExponentialProfile.State(1.05, 0);
+    pivotSetpoint = new ExponentialProfile.State(1.05, 0);
+  }
+
+  /** Updates the set of loggable inputs. */
+  @Override
+  public void updateInputs(PivotIOInputs inputs) {
+    if (!openLoop) {
+      double currentVelocity = pivotSetpoint.velocity;
+      pivotSetpoint = pivotProfile.calculate(0.02, pivotSetpoint, pivotGoal);
+      double feedForward =
+          pivotFFModel.calculate(
+              pivotSetpoint.position,
+              pivotSetpoint.velocity,
+              (pivotSetpoint.velocity - currentVelocity) / 0.02);
+      appliedVolts =
+          MathUtil.clamp(
+              controller.calculate(absoluteEncoder.getAbsolutePosition(), pivotSetpoint.position)
+                  + feedForward,
+              -12.0,
```

Comment:
More magic numbers. Why is voltage -12.0 though? Shouldn't it be 0?

Reply:
negative voltage = backwards spin so range is -12V to 12V